src/core/nixl_plugin_manager.cpp

```diff
@@ -358,36 +208,30 @@ nixl_mem_list_t nixlPluginHandle::getBackendMems() const {
     return mems; // Return empty mems if not implemented
 }
 
-std::vector<nixl_backend_t> nixlPluginManager::getLoadedPluginNames() {
-    lock_guard lg(lock);
-
-    std::vector<nixl_backend_t> names;
-    for (const auto& pair : loaded_plugins_) {
-        names.push_back(pair.first);
-    }
-    return names;
-}
-
 void nixlPluginManager::registerStaticPlugin(const char* name, nixlStaticPluginCreatorFunc creator) {
-    lock_guard lg(lock);
-
     nixlStaticPluginInfo info;
     info.name = name;
     info.createFunc = creator;
-    static_plugins_.push_back(info);
+    staticPlugins_.push_back(info);
 
     //Static Plugins are considered pre-loaded
     nixlBackendPlugin* plugin = info.createFunc();
     NIXL_INFO << "Loading static plugin: " << name;
     if (plugin) {
-        // Register the loaded plugin
-        auto plugin_handle = std::make_shared<const nixlPluginHandle>(nullptr, plugin);
-        loaded_plugins_[name] = plugin_handle;
+        // Register the loaded plugin (nullptr handle for static plugins)
+        dlHandleDeleter deleter(""); // No cleanup for static plugins
+        std::unique_ptr<void, dlHandleDeleter> handle(nullptr, deleter);
+        auto plugin_handle = createPluginHandle(std::move(handle), plugin);
+
+        // Store in base class map
+        lock_guard lg(lock_);
+        loadedPlugins_.emplace(name, plugin_handle);
```

Comment:
I think it would be better to put this into a protected method of a base class, so `loadedPlugins_` and `lock_` can be moved to private section.